healthz/handlers/healthz.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+	"github.com/gravitational/satellite/healthz/defaults"
+)
+
+// Auth checks if client supplied proper access key, if not sends
+// HTTP 401 Unauthorized and return false, otherwise returns true
```

Comment:
`Auth checks if client supplied proper access key and responds with HTTP 401 if authorization fails. Returns true if authorization was successful`